### PR TITLE
fix: gitlint workflow runs on pull request only

### PR DESCRIPTION
**Pull request by Omeramsc**
# Description

Separates gitlint from the workflow for pull-request and push, as it cannot and should not run on push.

The new workflow runs it only for the pull-request process itself.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added meaningful description with JIRA/GitHub issue key (if applicable)
- [ ] I have updated labels (if needed)

### CODE REVIEW

File: .github/workflows/pull_request.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-name: checks on Pull Request
+name: checks on Pull Request and push
```
Comment (yftacherzog) [inline .github/workflows/pull_request.yaml:2]: I think we should also rename the file

**Review by ifireball — APPROVED** [review]

File: .github/workflows/pull_request.yaml (hunk 1 above)
Comment (Omeramsc) [reply .github/workflows/pull_request.yaml]: done

**Review by yftacherzog — COMMENTED** [review]
I think we should decide whether we name the workflow  based on what they are doing or based on when they are running.

Accordingly, either the gitlint file can be renamed to pull_request, or the pull_request_and_push should be named according to what we're doing there.

**Comment by Omeramsc** [comment]
> I think we should decide whether we name the workflow based on what they are doing or based on when they are running.
> 
> Accordingly, either the gitlint file can be renamed to pull_request, or the pull_request_and_push should be named according to what we're doing there.

Do we have further plans for checks to run during pull request **only**? 
pull_request_and_push will include more checks in the future, it cannot really include all of them, unless we call it "code quality" or something like that.

**Review by yftacherzog — APPROVED** [review]
> > I think we should decide whether we name the workflow based on what they are doing or based on when they are running.
> > Accordingly, either the gitlint file can be renamed to pull_request, or the pull_request_and_push should be named according to what we're doing there.
> 
> Do we have further plans for checks to run during pull request **only**? pull_request_and_push will include more checks in the future, it cannot really include all of them, unless we call it "code quality" or something like that.

Don't know. I'll let you decide how to go about this. I was initially against the inconsistent naming convention, but it might make sense. 

lgtm

**Pull request merged**